prov/efa/src/rxr/rxr_pkt_type_base.c

```diff
@@ -260,8 +267,28 @@ ssize_t rxr_pkt_copy_data_to_rx_entry(struct rxr_ep *ep,
 
 	pkt_entry->x_entry = rx_entry;
 
-	if (data_size == 0)
-		goto out;
+	/*
+	 * Under 3 rare situations, this function does not perform the copy
+	 * action, but still consider data is copied:
+	 *
+	 * 1. application cancelled the receive, thus the receive buffer is not
+	 *    available to copy. In the case, this function is still called because
+	 *    receiving end point did not notify the sender about the cancelation,
+	 *    thus sender will keep sending.
+	 *
+	 * 2. application's receiving buffer is smaller than incoming message size,
+	 *    and this function will not copy to beyond receiving buffer.
+	 *    Again, in the case, receiving end point did not notify the sender
+	 *    about the truncation, thus sender will keep sending.
+	 *
+	 * 3. message size is 0, thus not data to copy
+	 */
+	if (OFI_UNLIKELY((rx_entry->rxr_flags & RXR_RECV_CANCEL)) ||
+	    OFI_UNLIKELY(data_offset >= rx_entry->cq_entry.len) ||
+	    OFI_UNLIKELY(data_size == 0)) {
+		rxr_pkt_handle_data_copied(ep, pkt_entry, data_size);
```

Comment:
Why not keep the `out:` section and `goto out`?

Reply:
I generally try to avoid using goto.